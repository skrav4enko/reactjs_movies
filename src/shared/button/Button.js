import React from 'react';
import './Button.scss';

function Button(props) {
  const { children, type = 'button', variant, size, onClick } = props;

  return (
    <button className={`button ${variant} ${size}`} type={type} onClick={onClick}>
      {children}
    </button>
  );
}

export default Button;
