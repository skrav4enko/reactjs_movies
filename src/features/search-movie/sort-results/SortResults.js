import React from 'react';
import './SortResults.scss';
import Button from '../../../shared/button/Button';
import ButtonGroup from '../../../shared/button-group/ButtonGroup';

function SortResults(props) {
  const { describe } = props;
  return (
    <div className="sort-results">
      <div className="container">
        <div className="sort-results__count">{describe}</div>
        <div className="sort-results__filter">
          <span className="sort-results__filter-by">Sort by</span>
          <ButtonGroup>
            <Button size="small">Release date</Button>
            <Button variant="secondary" size="small">
              Rating
            </Button>
          </ButtonGroup>
        </div>
      </div>
    </div>
  );
}

export default SortResults;
