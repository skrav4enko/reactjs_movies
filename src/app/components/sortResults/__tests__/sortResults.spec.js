/* eslint-disable */
import React from 'react';
import { shallow, mount } from 'enzyme';
import SortResults from '..';
import Button from '../../../shared/button';

describe('Sort Results', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      describe: 'string',
      sortBy: 'release_date',
      handleSort: jest.fn(),
    };
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<SortResults {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleSort', () => {
    const component = mount(<SortResults {...defaultProps} />);
    const button = component.find('.button.secondary').simulate('click');
    expect(defaultProps.handleSort).toHaveBeenCalledWith('vote_average');
  });

  it('should call handleSort', () => {
    const testProps = {
      ...defaultProps,
      sortBy: 'vote_average'
    };

    const component = mount(<SortResults {...testProps} />);
    const button = component.find('.button.secondary').simulate('click');
    expect(testProps.handleSort).toHaveBeenCalledWith('release_date');
  });
});

