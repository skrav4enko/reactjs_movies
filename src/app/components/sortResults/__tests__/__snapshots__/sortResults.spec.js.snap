// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort Results should match snapshot 1`] = `
<index.tsx
  describe="string"
  handleSort={[MockFunction]}
  sortBy="release_date"
/>
`;
